lib/analysis/modules/tests/ies_iteration.cpp

```diff
@@ -0,0 +1,235 @@
+#include <ert/util/test_util.hpp>
+#include <ert/util/rng.h>
+
+#include <ert/res_util/es_testdata.hpp>
+
+#include <ert/analysis/std_enkf.hpp>
+#include "ies_enkf_data.h"
+#include "ies_enkf_config.h"
+#include "ies_enkf.h"
+
+
+void init_stdA(const res::es_testdata& testdata, matrix_type * A2) {
+  rng_type * rng = rng_alloc( MZRAN, INIT_DEFAULT );
+  std_enkf_data_type * std_data = static_cast<std_enkf_data_type*>(std_enkf_data_alloc());
+  std_enkf_set_truncation(std_data, 1.00);
+  matrix_type * X = matrix_alloc(testdata.active_ens_size, testdata.active_ens_size);
+
+  std_enkf_initX(std_data,
+                 X,
+                 nullptr,
+                 testdata.S,
+                 testdata.R,
+                 testdata.dObs,
+                 testdata.E,
+                 testdata.D,
+                 rng);
+
+  matrix_inplace_matmul(A2, X);
+
+  std_enkf_data_free(std_data);
+  rng_free(rng);
+}
+
+
+void cmp_std_ies(const res::es_testdata& testdata) {
+  int num_iter = 100;
+  rng_type * rng = rng_alloc( MZRAN, INIT_DEFAULT );
+  matrix_type * A1 = testdata.alloc_state("prior");
+  matrix_type * A2 = testdata.alloc_state("prior");
+  matrix_type * S  = testdata.alloc_state("S");
```

Comment:
This has given you a matrix of the same size and shape as A - that is not correct; just use plain matrix_alloc

Reply:
yes, lots of new notation in these routines... fixed below.